js/str.js

```diff
@@ -1,14 +1,217 @@
+// Partially from
+//  https://github.com/ejgallego/jscoq/blob/2718f9caf31398704c2d84ff089e3f6f0321eada/coq-js/js_stub/str.js
```

Comment:
this is AGPL-3+, and the part based on strstubs.c is LGPL2 + static linking exception. we will have to review license compatibility...

without looking into how hard it'd be first, i'd be more inclined to remove the uses of `Str` instead. might even make sense to use a real parser (we already depend on sedlex and ocamllex) but that's a bigger project.

Reply:
Thanks, I'll remove it for now, I think LGPL2 is compatible with MIT, not sure about AGPL3

Reply:
https://github.com/ejgallego/jscoq/issues/97

Reply:
`Str` also would be useful for flow.js overall, not just flowconfig parser

Reply:
Js_of_ocaml (3.6) now include all full support for Str